CMakeLists.txt

```diff
@@ -49,6 +49,20 @@ if( ${TARGET_OS} STREQUAL "Linux" )
     message("Unsupported architecture: ${TARGET_ARCHITECTURE}" )
     return()
   endif()
+elseif( ${TARGET_OS} STREQUAL "OpenBSD")
+  set(OS "OPENBSD")
+  set(OSNAME "OpenBSD")
+  if ( ${TARGET_ARCHITECTURE} STREQUAL "amd64" )
+    set(ARCHNAME x86-64)
+    set(ARCH X86)
+    set(WRDSZ 64)
+    set(TARGET_ARCHITECTURE x86_64)
```

Comment:
When I was playing around with the port, I did the same thing -- set TARGET_ARCHITECTURE to x86_64.  I'm wondering if we shouldn't factor our TARGET_ARCHITECTURE to be something like FLANG_TARGET_ARCHITECTURE.  The downside is that it might introduce confusion and bugs because people know about TARGET_ARCHITECTURE.  This isn't a call to action; just a discussion point that probably better raised in the mailing list :)